middleware.ts

```diff
@@ -92,14 +112,51 @@ export function middleware(req: NextRequest) {
   // Transform filters into an object
   const filtersObj: Record<string, string[]> = {};
 
-  searchParams.forEach((value, key) => {
+  for (const [key, value] of searchParams.entries()) {
     const match = key.match(/^filters\[(.*?)\](?:\[\])?$/);
     if (match) {
       let filterKey = match[1];
       let filterValue = decodeURIComponent(value);
       if (filterKey === "rights" && filterValue === "pd") {
         filterValue = "publicDomain";
       }
+      switch (filterKey) {
```

Comment:
Would this work if you created a map lookup such as
```
const map = {
  placeTerm_mtxt_s: "place",
  genre_mtxt_s: "genre",
  typeOfResource_mtxt_s: "type",
  publisher_mtxt_s: "publisher",
  namePart_mtxt_s: "name",
  topic_mtxt_s: "topic",
  languageTerm_mtxt_s: "language",
  form_mtxt_s: "form",
};
```

and keep the additional logic updates? E.g.
```
if (filterKey === "rights" && filterValue === "pd") {
  filterValue = "publicDomain";
} else if (filterKey === "root-collection") {
  filterKey = "collection";
  filterValue = await getCollectionTitle(filterValue);
} else if (map[filterKey]) {
  filterKey = map[filterKey];
}
```

just an idea to group the filter key and their mappings together.